scripts/pylib/twister/twisterlib.py

```diff
@@ -2498,6 +2496,7 @@ def process(self, pipeline, done, message, lock, results):
                 if res.get('returncode', 1) > 0:
                     pipeline.put({"op": "report", "test": self.instance})
                 else:
+                    self.suite.gather_footprints(inst)
```

Comment:
instead of putting this in the build section, create a new operation and let the pipeline deal with that. Also move the function to collect footprint data to the same class where it is being called.

Reply:
I updated the PR with your suggestion. Did I got them correctly?